pandas/_libs/tslib.pyx

```diff
@@ -1405,6 +1404,11 @@ cdef class _Timestamp(datetime):
         def __get__(self):
             return np.datetime64(self.value, 'ns')
 
+    def timestamp(self):
+        """Return POSIX timestamp as float."""
+        # py27 compat, see GH#17329
+        return self.value / 1e9
```

Comment:
I think this is wrong if we have a nanosecond component. should issue a warning if we have one (I think its a print statement, but that's consistent with what we have now). add a test for this.

Reply:
Why is it wrong with a nanosecond component?  Is it a precision issue?

Reply:
yes its not compatible with datetime.timestamp() (since you cannot construct one in datetime, we should not allow a time value which is invalid)

Reply:
I'll make this change, will add follow-up to the tslibs todo list, since it isn't obvious to me that this is the correct decision.